__tests__/json_decode_test.ml

```diff
@@ -472,49 +387,27 @@ describe "map" (fun () ->
   let open Json in
   let open! Decode in
 
-  test "boolean" (fun () ->
-    expectFn (int |> map ((+)2)) (Encode.boolean Js.true_) |> toThrow);
-  test "float" (fun () ->
-    expectFn (int |> map ((+)2)) (Encode.float 1.23) |> toThrow);
+  Test.throws (int |> map ((+)2)) [Bool; Float; String; Null; Array; Object];
+
   test "int" (fun () ->
     expect @@ (int |> map ((+)2)) (Encode.int 23) |> toEqual 25);
-  test "string" (fun () ->
-    expectFn (int |> map ((+)2)) (Encode.string "test") |> toThrow);
-  test "null" (fun () ->
-    expectFn (int |> map ((+)2)) (Encode.null) |> toThrow);
-  test "array" (fun () ->
-    expectFn (int |> map ((+)2)) (Encode.array [||]) |> toThrow);
-  test "object" (fun () ->
-    expectFn (int |> map ((+)2)) (Encode.object_ @@ Js.Dict.empty ()) |> toThrow);
 );
 
 describe "andThen" (fun () ->
   let open Json in
   let open! Decode in
 
-  test "boolean -> int" (fun () ->
-    expectFn (int |> andThen (fun _ -> int)) (Encode.boolean Js.true_) |> toThrow);
-  test "float -> int" (fun () ->
-    expectFn (int |> andThen (fun _ -> int)) (Encode.float 1.23) |> toThrow);
+  Test.throws (int |> andThen (fun _ -> int)) [Bool; Float; String; Null; Array; Object];
+  Test.throws (float |> andThen (fun _ -> int)) [Float];
+  Test.throws (int |> andThen (fun _ ->float)) [Float];
```

Comment:
Out of order. And there would be three tests named just "float" here, re: my comment about differentiating the names above.